com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -910,10 +963,29 @@ private void AddInterpolatedState(NetworkTransformState newState)
                 }
 
                 currentRotation.eulerAngles = currentEulerAngles;
+
                 m_RotationInterpolator.AddMeasurement(currentRotation, sentTime);
             }
         }
 
+        /// <summary>
+        /// Stops extrapolating the <see cref="m_LastReceivedState"/>.
+        /// </summary>
+        /// <remarks>
+        /// <see cref="OnNetworkStateChanged"/>
+        /// </remarks>
+        private void StopExtrapolatingLastState()
+        {
+            if (!m_LastReceivedState.IsDirty || m_LastReceivedState.EndExtrapolationTick >= NetworkManager.LocalTime.Tick)
```

Comment:
So does this mean if we haven't received a tick's data for more than a tick, we end the extrapolation? Cause extrapolation is also useful for more than 1 tick. The limit is actually 3x a tick.

Reply:
If you look at the original code (line 503) you will see the following:
`else if (!m_HasSentLastValue && m_CachedNetworkManager.LocalTime.Tick >= m_LastSentTick + 1)`
I believe the 3x a tick wiggle room was due to the fact that authority had to send the message to non-authority instances which could take (on average) between 1 and 2 ticks (with the additional tick coming from the m_LastSentTick + 1).
One of the issues with this approach typically involves scenarios where clients with high latencies (128ms+) could take 4+ ticks (TickRate of 30) to just receive the end extrapolation message, which at that point it could be extrapolated beyond the target value(s).  

What this approach does (handling the end extrapolation 1 tick after the state in question is received) is it removes the extra message (and associated latencies), it assures that it will interpolate up to but not beyond the state's values that changed (yeah it currently does make extrapolation sort of mute).  Then there is a second part of this fix where there is additional code (starting around line 680 the comments describe it in further detail) that keeps interpolating even if the current state has no deltas for an axial value.  This assures that if it does take longer than 1 tick to reach the target value it will still reach the target value (unless a new state is applied that overrides the values being interpolated towards).

Pro's and con's with both approaches...the issue I ran into with additively combining the m_LastSentState with any new state before the end of the current tick and then sending that at the start of a new tick involved the sent time (i.e. when received and adding the measurements it requires the time it was sent...but if it is multiple states combined ( following an "EITHER OR" pattern) then there is just one send state.  So... it was either creating a new message that would send multiple states (with all of that additional overhead) to end extrapolation for each currently in progress state or it was handling it on the non-authoritative side by just creating a copy of a newly received state and applying it a tick later.

Let me see if I can increase the tick offset without impacting anything.

Reply:
Yeah... that won't work out properly for this iteration.
We need to revisit how extrapolation is handled while using Quaternion.SlerpUnclamped.